harvester_e2e_tests/integration/test_networks.py

```diff
@@ -0,0 +1,954 @@
+            f"Still got {code} with {data}"
+        )
+
+
+def check_vm_running(api_client, unique_name, wait_timeout):
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+
+    while endtime > datetime.now():
+        code, data = api_client.vms.get(unique_name)
+        vm_fields = data['metadata']['fields']
+
+        assert 200 == code, (code, data)
+        if vm_fields[2] == 'Running':
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"Failed to create VM {unique_name} in Running status, exceed given timeout\n"
+            f"Still got {code} with {data}"
+        )
+
+
+def check_vm_ip_exists(api_client, unique_name, wait_timeout):
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+
+    while endtime > datetime.now():
+        code, data = api_client.vms.get_status(unique_name)
+        assert 200 == code, (f"Failed to get specific vm content: {code}, {data}")
+        if 'ipAddress' in data['status']['interfaces'][0]:
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"Failed to get VM {unique_name} IP address, exceed the given timed out\n"
+            f"Still got {code} with {data}"
+        )
+
+
+def delete_vm(api_client, unique_name, wait_timeout):
+    api_client.vms.delete(unique_name)
+
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+
+    while endtime > datetime.now():
+        code, data = api_client.vms.get_status(unique_name)
+        if code == 404:
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"Failed to delete VM {unique_name}, exceed given timeout\n"
+            f"Still got {code} with {data}"
+        )
+
+
+@pytest.mark.p0
+class TestBackendNetwork:
+
+    @pytest.mark.p0
+    @pytest.mark.dependency(name="mgmt_network_connection")
+    def test_mgmt_network_connection(self, api_client, request, client, image_opensuse,
+                                     unique_name, wait_timeout):
+        """
+        Manual test plan reference:
+        https://harvester.github.io/tests/manual/network/validate-network-management-network/
+
+
+        Steps:
+        1. Create a new VM
+        2. Make sure that the network is set to the management network with masquerade as the type
+        3. Wait until the VM boot in running state
+        4. Check can ping VM with management network from Harvester node
+        5. Check can SSH to VM with management network from Harvester node
+        6. Check can't SSH to VM with management network from external host
+        """
+        vip = request.config.getoption('--endpoint').strip('https://')
+
+        node_password = request.config.getoption("--host-password")
+
+        # Check image exists
+        code, data = api_client.images.get(image_opensuse.name)
+
+        if code == 404:
+            create_image_url(api_client, image_opensuse.name, image_opensuse.url, wait_timeout)
```

Comment:
I think what I'm saying is, use a fixture like:

https://github.com/harvester/tests/blob/cd34f0e5006d466d467ab359ac3e26e5b6f0d7e4/harvester_e2e_tests/integration/test_vm_functions.py#L74-L99

to make the image be the part of setup.